elasticgraph-graphql/lib/elastic_graph/graphql/aggregation/term_grouping.rb

```diff
@@ -31,6 +31,7 @@ def composite_clause(grouping_options: {})
 
         def non_composite_clause_for(query)
           clause_value = work_around_elasticsearch_bug(terms_subclause)
+          clause_value["missing"] = missing_value_placeholder if handles_missing_values?
```

Comment:
```suggestion
          clause_value = clause_value.merge({"missing" => missing_value_placeholder}) if handles_missing_values?
```

As a general rule, we favor a coding style that avoids mutating objects unless strictly necessary. Re-assigning a local variable to a different object is fine though--local variables, by their nature, are localized.